inception-search-mtas/src/main/java/de/tudarmstadt/ukp/inception/search/index/mtas/MtasUimaParser.java

```diff
@@ -242,68 +242,158 @@ else if (annotation instanceof Sentence) {
+                                    indexedStr = IndexingConstants.kbEntity + MtasToken.DELIMITER
+                                            + kbValue;
+                                    log.debug("Indexed String without type and label for : {}", indexedStr);
+                                    MtasToken mtasAnnotationKBEntity = new MtasTokenString(
+                                            mtasId++, indexedStr, beginToken);
+                                    mtasAnnotationKBEntity.setOffset(annotation.getBegin(),
+                                            annotation.getEnd());
+                                    mtasAnnotationKBEntity.addPositionRange(beginToken,
+                                            endToken);
+                                    tokenCollection.add(mtasAnnotationKBEntity);
+                                    
+                                }
+
                             }
                         }
                     }
                 }
             }
-        }
-        catch (Exception e) {
+        } catch (Exception e) {
             log.error("Unable to index document", e);
         }
         return tokenCollection;
     }
-
+    
     @Override
-    public String printConfig()
-    {
+    public String printConfig() {
         return null;
     }
 
-    class SimpleAnnotationLayer
-    {
+    /**
+     * Takes in IRI for identifier and returns teh label String Eg: InputParameter :-
+     * http://www.w3.org/TR/2003/PR-owl-guide-20031209/wine#RoseDAnjou Returned :-
+     * "KBConcept"+MtasToken.DELIMITER+RoseDAnjou
+     * 
+     * @param iri
+     * @return
+     */
+    public String getUILabel(String aIRI) {
+
+        StringBuilder labelStr = new StringBuilder();
+        System.out.println();
+        Optional<KBObject> kbObject = kbUtil.readKBIdentifier(project, aIRI);
+        if (kbObject.isPresent()) {
+            labelStr.append(kbObject.get().getClass().getSimpleName() + MtasToken.DELIMITER
+                    + kbObject.get().getUiLabel());
+        } else {
+            return labelStr.toString();
+        }
+
+        return labelStr.toString();
+    }
+    
+    /**
+     * Method Implementation to get MtasToken (To be done when subclass and class 
+     * semantics to be included
+     * @return
+     */
+    public MtasToken getMtasTokenKBFeature() {
+        return null;
+        
+    }
+
+    class SimpleAnnotationLayer {
         private ArrayList<Pair<String, String>> features;
         private String LayerName;
         private String LayerUiName;
         private String LayerShorName;
 
-        public ArrayList<Pair<String, String>> getFeatures()
-        {
+        public ArrayList<Pair<String, String>> getFeatures() {
             return features;
         }
 
-        public void setFeatures(ArrayList<Pair<String, String>> features)
-        {
+        public void setFeatures(ArrayList<Pair<String, String>> features) {
```

Comment:
Use `List` instead of `ArrayList` when declaring fields, parameters, or return values.

Argument names should always start with an `a`, so `aFeatures` instead of `features`. Applies everywhere.

Reply:
`SimpleAnnotationLayer` has not been used. Can this be removed?

Reply:
If it is not used, I guess it can be removed. @betoboullosa ?

Reply:
@naveen2507 so what about the SimpleAnnotationLayer now? If it is not used and @betoboullosa does not respond, then please remove it.